src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs

```diff
@@ -108,7 +109,7 @@ public CSharpInlineTypeHintsService()
             // if this is a hint that is placed in-situ (i.e. it's not overwriting text like 'var'), then place
             // a space after it to make things feel less cramped.
             var trailingSpace = span.Length == 0;
-            return new TypeHint(type, span, trailingSpace: trailingSpace);
+            return new TypeHint(type, span, new TextChange(displayAllSpan.Span, type.ToDisplayString(s_minimalTypeStyle)), trailingSpace: trailingSpace);
```

Comment:
it would be good to have a test to ensure that double clickong on a non-overwrite hint overwrites teh right text.

Reply:
How would you go about testing that? The testing suite we currently have has access to the InlineHInts, but not the tags themselves. If we had access to the tags we could invoke a double click on the adornment, but is there an easier way to do it? @CyrusNajmabadi

Reply:
well, we can have two sorts of tests.  Unit tests that would validate our internal API and hte data it returns.  Then an integration test that actually loads this into VS and interacts with the real tags on the editor side.  

We should likely have both.  Most tests should be at teh unit level, but we should have a couple of tests that actually do the full end to end validation at the editor level.

Reply:
this code feels like it should be in the constructor of TypeHint.  Pass both spans.